sys/config/src/config.c

```diff
@@ -194,6 +196,34 @@ conf_value_from_str(char *val_str, enum conf_type type, void *vp, int maxlen)
         }
         *(int64_t *)vp = val64;
         break;
+    case CONF_UINT8:
+    case CONF_UINT16:
+    case CONF_UINT32:
+        uval = strtoul(val_str, &eptr, 0);
+        if (*eptr != '\0') {
+            goto err;
+        }
+        if (type == CONF_UINT8) {
+            if (uval > UINT8_MAX) {
+                goto err;
+            }
+            *(uint8_t *)vp = uval;
+        } else if (type == CONF_UINT16) {
+            if (uval > UINT16_MAX) {
+                goto err;
+            }
+            *(uint16_t *)vp = uval;
+        } else if (type == CONF_UINT32) {
+            *(uint32_t *)vp = uval;
+        }
+        break;
+    case CONF_UINT64:
+        uval64 = strtoull(val_str, &eptr, 0);
```

Comment:
I see there's special handling for `UINT64` here and also for `INT64` above, but I am not sure why you can't just use `strtoll` for all of them.

Reply:
I agree.  My guess is that the "ll" versions of these functions felt too heavy weight to use for the <=32-bit types.  If I were implementing it from scratch, I think I would just use the "ll" functions across the board.  I am reluctant to change this existing code though... this feels like an easy place to introduce a bug.

Reply:
It looks like both `strtoul` and `strtoull` are mapped to `strntoumax` inside baselibc, just using a different cast!

Reply:
OK @utzig, you convinced me to change everything :).  I also noticed `conf_value_from_str()` had a bug in its range checks for signed integers, so I have fixed that.

Would you mind taking a look at the latest commits?

Reply:
> Would you mind taking a look at the latest commits?

Just did, looks good!